modules/common/tests/src/Unit/DataResourceTest.php

```diff
@@ -82,4 +84,47 @@ public function testGetIdentifierAndVersionException() {
     $result = DataResource::getIdentifierAndVersion($id);
   }
 
+  /**
+   * @covers ::createNewPerspective
+   */
+  public function testCreateNewPerspective() {
+    $data_resource = new DataResource(
+      '/foo/bar',
+      'txt',
+      DataResource::DEFAULT_SOURCE_PERSPECTIVE
+    );
+
+    $clone_data_resource = $data_resource->createNewPerspective('local_url', 'uri://foo/bar');
+
+    // Not the same object.
+    $this->assertNotSame($data_resource, $clone_data_resource);
+    // Clone contains 'local_url' perspective.
+    $ref_perspective = new \ReflectionProperty($clone_data_resource, 'perspective');
+    $ref_perspective->setAccessible(TRUE);
+    $this->assertEquals('local_url', $ref_perspective->getValue($clone_data_resource));
+  }
+
+  /**
+   * @covers ::createNewVersion
+   */
+  public function testCreateNewVersion() {
+    $data_resource = new DataResource(
+      '/foo/bar',
+      'txt',
+      DataResource::DEFAULT_SOURCE_PERSPECTIVE
+    );
+
+    $clone_data_resource = $data_resource->createNewVersion();
+
+    // Not the same object.
+    $this->assertNotSame($data_resource, $clone_data_resource);
+    // Clone contains new version.
+    $ref_version = new \ReflectionProperty(DataResource::class, 'version');
```

Comment:
Same question - why not use `getVersion()`?

Reply:
Fixed.